include/swift/Basic/LangOptions.h

```diff
@@ -41,6 +42,9 @@ namespace swift {
     /// Language features
     ///
 
+    /// \brief User-overridable language version to compile for.
+    version::Version EffectiveLanguageVersion = version::Version::getCurrentLanguageVersion();
```

Comment:
Can we put this into `ModuleDecl`? It's plausible that, e.g., the optimizer might want to have multiple modules within the same `ASTContext`, and those modules might be built with different language modes.